css/src/atomics/spacing.scss

```diff
@@ -21,211 +20,66 @@ $layout-sizes: (
-				padding-#{$direction}: sizeValue($sizeKey, $sizeValue) !important;
+@include tablet {
+	@each $property in $spacing-properties {
+		// iterates through margin, margin-inline, margin-block, etc
+		@each $size in $layout-sizes {
+			// iterates over values 'xs', 's', 'l' ... 'none'
+			$sizeKey: nth($size, 1);
+			$sizeValue: nth($size, 2);
+
+			// .<property>-<value>-tablet
+			.#{$property}#{$separator}#{$sizeKey}#{$separator}tablet {
+				// margin-xs-tablet, margin-s-tablet, etc.
+				#{$property}: $sizeValue !important;
 			}
 		}
 	}
 }
 
 @include desktop {
-	@each $size in $layout-sizes {
-		$sizeKey: nth($size, 1);
-		$sizeValue: nth($size, 2);
-
-		.#{$marginKey}#{$separator}#{$sizeKey}-desktop {
-			margin: sizeValue($sizeKey, $sizeValue) !important;
-		}
-
-		.#{$paddingKey}#{$separator}#{$sizeKey}-desktop {
-			padding: sizeValue($sizeKey, $sizeValue) !important;
-		}
-
-		// margin-inline and margin-block property isn't supported by IE, Edge, and Safari yet (as of Apr 2021)
-
-		.#{$marginKey}#{$separator}#{$inlineKey}#{$separator}#{$sizeKey}-desktop {
-			// Replace with margin-inline property when possible
-			margin-right: sizeValue($sizeKey, $sizeValue) !important;
-			margin-left: sizeValue($sizeKey, $sizeValue) !important;
-		}
-
-		.#{$paddingKey}#{$separator}#{$inlineKey}#{$separator}#{$sizeKey}-desktop {
-			// Replace with margin-inline property when possible
-			padding-right: sizeValue($sizeKey, $sizeValue) !important;
-			padding-left: sizeValue($sizeKey, $sizeValue) !important;
-		}
-
-		.#{$marginKey}#{$separator}#{$blockKey}#{$separator}#{$sizeKey}-desktop {
-			// Replace with margin-block property when possible
-			margin-top: sizeValue($sizeKey, $sizeValue) !important;
-			margin-bottom: sizeValue($sizeKey, $sizeValue) !important;
-		}
-
-		.#{$paddingKey}#{$separator}#{$blockKey}#{$separator}#{$sizeKey}-desktop {
-			// Replace with margin-block property when possible
-			padding-top: sizeValue($sizeKey, $sizeValue) !important;
-			padding-bottom: sizeValue($sizeKey, $sizeValue) !important;
-		}
-
-		@each $position-key, $direction in $directions {
-			.#{$marginKey}#{$separator}#{$position-key}#{$separator}#{$sizeKey}-desktop {
-				margin-#{$direction}: sizeValue($sizeKey, $sizeValue) !important;
-			}
-
-			.#{$paddingKey}#{$separator}#{$position-key}#{$separator}#{$sizeKey}-desktop {
-				padding-#{$direction}: sizeValue($sizeKey, $sizeValue) !important;
+	@each $property in $spacing-properties {
+		// iterates through margin, margin-inline, margin-block, etc
+		@each $size in $layout-sizes {
+			// iterates over values 'xs', 's', 'l' ... 'none'
+			$sizeKey: nth($size, 1);
+			$sizeValue: nth($size, 2);
+
+			// .<property>-<value>-desktop
+			.#{$property}#{$separator}#{$sizeKey}#{$separator}desktop {
+				// margin-xs-desktop, margin-s-desktop, etc.
+				#{$property}: $sizeValue !important;
 			}
 		}
 	}
 }
-
 @include widescreen {
```

Comment:
super nit: add an empty row before that one :)

Reply:
Ah good eye, updated